### PR TITLE
add sort by name to assigned page

### PR DESCRIPTION
**Pull request by nickvisut**
This PR adds sorting functionality on the front end for the Assigned page. It does so by adding a new button (simple stand-in for current placeholder) that toggles between name ascending and descending sort. It partly addresses #424.

Note that each interaction currently forces a rerender, so we will want to revisit in future to improve perf (through either `useEffect()` and/or server-side sorting). This is good enough for POC, however.

Also note that we will likely need to replace the button with a sorting dropdown to incorporate other sort properties.

### CODE REVIEW

**Comment by nickvisut** [comment]
rebased on main to get around linting issue

File: src/pages/benchmarks/index.tsx [hunk 1]
```diff
@@ -12,21 +12,50 @@ import $button from "../../components/design_system/button/Button.module.css";
...
+type SortDirection = "asc" | "desc";
+
 function Benchmarks() {
   const [isPara, setIsPara] = useState(false);
-  const { data: tasks, isLoading } = trpc.para.getMyTasks.useQuery();
+
+  const [sortProperty, setSortProperty] = useState<SortProperty>("first_name");
+  const [sortDirection, setSortDirection] = useState<SortDirection>("asc");
+
+  const { data: tasksData, isLoading, error } = trpc.para.getMyTasks.useQuery();
```
Comment (katconnors) [inline src/pages/benchmarks/index.tsx:24]: Is "error" within data going to be accessed, or is this for troubleshooting of some sort?

File: src/pages/benchmarks/index.tsx [hunk 2]
```diff
@@ -12,21 +12,50 @@ import $button from "../../components/design_system/button/Button.module.css";
...
 
   const handleTogglePara = () => {
     setIsPara(!isPara);
   };
 
+  const getSortedTasks = () => {
+    if (!tasksData) return [];
+    return [...tasksData].sort((a, b) => {
+      if (a[sortProperty] < b[sortProperty])
+        return sortDirection === "asc" ? -1 : 1;
```
Comment (katconnors) [inline src/pages/benchmarks/index.tsx:34]: I had a bit of trouble following why you want to return 1 or -1. Could you please explain? Thanks!

**Review by katconnors — COMMENTED** [review]
Just had a couple of comments, code wise. 
For testing the front end functionality, did you somehow swap to a "para" role?

File: src/pages/benchmarks/index.tsx (hunk 2 above)
Comment (canjalal) [reply src/pages/benchmarks/index.tsx:34]: JavaScript decides whether two items in a sort are in the correct order relative to another depending on whether the sorting callback you give it returns +1 or -1.

File: src/pages/benchmarks/index.tsx (hunk 1 above)
Comment (canjalal) [reply src/pages/benchmarks/index.tsx:24]: We can remove it for now

File: src/pages/benchmarks/index.tsx (hunk 2 above)
Comment (katconnors) [reply src/pages/benchmarks/index.tsx:34]: ty!

File: src/pages/benchmarks/index.tsx (hunk 1 above)
Comment (katconnors) [reply src/pages/benchmarks/index.tsx:24]: sounds good

**Comment by canjalal** [comment]
> Just had a couple of comments, code wise. For testing the front end functionality, did you somehow swap to a "para" role?

Yes, until we add admin functionality to impersonate users, we have to go into the database and flip our role to "para"

File: src/pages/benchmarks/index.tsx [hunk 3]
```diff
@@ -11,22 +11,50 @@ import { useState } from "react";
...
+  const { data: tasksData, isLoading } = trpc.para.getMyTasks.useQuery();
 
   const handleTogglePara = () => {
     setIsPara(!isPara);
   };
 
+  const getSortedTasks = () => {
+    if (!tasksData) return [];
+
+    return [...tasksData].sort((a, b) => {
```
Comment (thomhickey) [inline src/pages/benchmarks/index.tsx:31]: sorting a copy of the array bypasses reactivity for tasksData, is this intentional?

File: src/pages/benchmarks/index.tsx [hunk 4]
```diff
@@ -11,22 +11,50 @@ import { useState } from "react";
...
+  const handleSort = (property: SortProperty) => {
+    if (property === sortProperty) {
+      setSortDirection(sortDirection === "asc" ? "desc" : "asc");
+    } else {
+      setSortProperty(property);
+      setSortDirection("asc");
+    }
+  };
+
+  const displayedTasks = getSortedTasks();
```
Comment (thomhickey) [inline src/pages/benchmarks/index.tsx:49]: maybe just sort and display tasksData? no need for displayedTasks?

**Review by francisli — APPROVED** [review]

**Pull request merged**